plugins/modules/nd_rest.py

```diff
@@ -250,7 +257,11 @@ def main():
     method = nd.params.get("method").upper()
 
     # Append previous state of the object
-    if method in ("PUT", "DELETE", "PATCH"):
+    # Purpose of ignore_previous_state:
+    # The ignore_previous_state was introduced because querying existing objects via the GET method on the /api/config/routes endpoint is not supported.
+    # To avoid the /api/config/routes GET call ignore_previous_state was introduced.
+    # Additionally, POST and DELETE methods return None for the /api/config/routes.
+    if method in ("PUT", "DELETE", "PATCH") and not ignore_previous_state:
```

Comment:
Why filter the methods here?

Reply:
The methods are already part of the legacy code. To maintain the existing behaviour, I keep all the filter methods.